### PR TITLE
Updating gloo-mesh-demo to GM 2.1.0-beta29 and Istio 1.15.1

### PR DESCRIPTION
**Pull request by nickschuetz**

### CODE REVIEW

File: workshops/gloo-mesh-demo/README.md [hunk 1]
```diff
@@ -46,20 +46,20 @@ To get started with this workshop, clone this repo.
...
 
 # Istio version information
 export ISTIO_IMAGE_REPO=us-docker.pkg.dev/gloo-mesh/istio-workshops
-export ISTIO_IMAGE_TAG=1.13.5-solo
-export ISTIO_VERSION=1.13.5
+export ISTIO_IMAGE_TAG=1.15.1-solo
+export ISTIO_VERSION=1.15.1
 ```
 
 ## Lab 1 - Configure/Deploy the Kubernetes clusters <a name="Lab-1"></a>
```
Comment (nmnellis) [inline workshops/gloo-mesh-demo/README.md:65]: Can you add `meshctl check` after the agents are installed?

File: workshops/gloo-mesh-demo/README.md [hunk 2]
```diff
@@ -154,7 +154,7 @@ export PATH=$PWD/istio-${ISTIO_VERSION}/bin:$PATH
 istioctl version
 ```
 
-2. Install Istio to each of the remote clusters. If you're using local Kubernetes clusters on a Mac M1 or M2, use [these ARM instructions](problems-istio-arm.md) instead.
```
Comment (nmnellis) [inline workshops/gloo-mesh-demo/README.md:157]: If arm isnt needed anymore, we can delete the `problems-istio-arm.md` files as well as the istiooperator files for arm

**Review by nmnellis — COMMENTED** [review]
I got stuck on the failover part, it appears that failover isnt working correctly and neither is outlier detection. Did it work for you?

**Comment by nickschuetz** [comment]
> 

This is my test from the run yesterday:

```
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % kubectl --context ${MGMT} apply -f - <<'EOF'
apiVersion: networking.gloo.solo.io/v2
kind: RouteTable
metadata:
  name: frontend
  namespace: web-team
  labels:
    lab: failover
spec:
  hosts:
    - '*'
  virtualGateways:
    - name: north-south-gw
      namespace: ops-team
      cluster: mgmt
  workloadSelectors: []
  http:
    - name: frontend
      labels:
        virtual-destination: frontend
        oauth: "true"
      forwardTo:
        destinations:
          - ref:
              name: frontend
              namespace: web-team
            kind: VIRTUAL_DESTINATION
            port:
              number: 80
EOF
routetable.networking.gloo.solo.io/frontend configured
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % kubectl --context ${MGMT} apply -f - <<'EOF'
apiVersion: resilience.policy.gloo.solo.io/v2
kind: FailoverPolicy
metadata:
  name: failover
  namespace: web-team
spec:
  applyToDestinations:
  - kind: VIRTUAL_DESTINATION
    selector:
      namespace: web-team
  config:
    # enable default locality based load balancing
    localityMappings: []
EOF
failoverpolicy.resilience.policy.gloo.solo.io/failover creatednschuetz@Nicks-MacBook-Pro gloo-mesh-demo % kubectl --context ${MGMT} apply -f - <<'EOF'
apiVersion: resilience.policy.gloo.solo.io/v2
kind: OutlierDetectionPolicy
metadata:
  name: outlier-detection
  namespace: web-team
spec:
  applyToDestinations:
  - kind: VIRTUAL_DESTINATION
    selector:
      namespace: web-team
  config:
    consecutiveErrors: 2
    interval: 5s
    baseEjectionTime: 15s
    maxEjectionPercent: 100
EOF
outlierdetectionpolicy.resilience.policy.gloo.solo.io/outlier-detection created
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % kubectl --context $CLUSTER1 -n web-ui patch deploy frontend --patch '{"spec":{"template":{"spec":{"containers":[{"name":"server","command":["sleep","20h"],"readinessProbe":null,"livenessProbe":null}]}}}}'
deployment.apps/frontend patched
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster2     <b>Locality:</b> us-east-2/us-east-2b</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % EOF

nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % kubectl --context $CLUSTER1 -n web-ui patch deploy frontend --patch '{"spec":{"template":{"spec":{"containers":[{"name":"server","command":[],"readinessProbe":null,"livenessProbe":null}]}}}}'
deployment.apps/frontend patched
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % kubectl --context $CLUSTER1 -n web-ui patch deploy frontend --patch '{"spec":{"template":{"spec":{"containers":[{"name":"server","command":[],"readinessProbe":null,"livenessProbe":null}]}}}}'
deployment.apps/frontend patched
nschuetz@Nicks-MacBook-Pro gloo-mesh-demo % for i in {1..6}; do curl -sSk http://$GLOO_GATEWAY | grep "Cluster Name:"; done
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
                <div class="h-free-shipping"><b>Cluster Name:</b> cluster1     <b>Locality:</b> us-east-2/us-east-2a</div>
```

So it appeared to look good to me.

**Comment by nmnellis** [comment]
I dont think we can use Istio 1.14 or 1.15 until this is fixed https://github.com/solo-io/gloo-mesh-enterprise/issues/5555

**Pull request merged**